test_util/test_upgrade_vpc.py

```diff
@@ -159,125 +132,279 @@ def wait_for_dns(hostname):
+        self.dcos_api_session_factory_upgrade = dcos_api_session_factory_upgrade
+        self.num_masters = num_masters
+        self.num_agents = num_agents
+        self.num_public_agents = num_public_agents
+        self.stable_installer_url = stable_installer_url
+        self.installer_url = installer_url
+        self.aws_region = aws_region
+        self.aws_access_key_id = aws_access_key_id
+        self.aws_secret_access_key = aws_secret_access_key
+        self.ssh_key = ssh_key
+        self.default_os_user = default_os_user
+        self.config_yaml_override_install = config_yaml_override_install
+        self.config_yaml_override_upgrade = config_yaml_override_upgrade
+
+        self.teamcity_msg = TeamcityServiceMessages()
+
+        # the two following properties are set when running setup_cluster_workload, here we default them to empty
+        # values.
+        self.test_app_ids = []
+        self.tasks_start = []
+
+    @staticmethod
+    def app_task_ids(dcos_api, app_id):
+        """Return a list of Mesos task IDs for app_id's running tasks."""
+        assert app_id.startswith('/')
+        response = dcos_api.marathon.get('/v2/apps' + app_id + '/tasks')
+        response.raise_for_status()
+        tasks = response.json()['tasks']
+        return [task['id'] for task in tasks]
+
+    def parse_dns_log(self, dns_log_content):
+        """Return a list of (timestamp, status) tuples from dns_log_content."""
+        dns_log = [line.strip().split(' ') for line in dns_log_content.strip().split('\n')]
+        if any(len(entry) != 2 or entry[1] not in ['SUCCESS', 'FAILURE'] for entry in dns_log):
+            message = 'Malformed DNS log.'
+            self.log.debug(message + ' DNS log content:\n' + dns_log_content)
+            raise Exception(message)
+        return dns_log
+
+    def log_test(self, test_name: str, call: Callable[[], None]) -> None:
+        try:
+            self.teamcity_msg.testStarted(test_name)
+            call()
+        except Exception:
+            # we want this except to be broad so that we can keep any Exception from taking
+            # everything with it and not asserting the other tests
+            self.teamcity_msg.testFailed(test_name, details=traceback.format_exc())
+        finally:
+            self.teamcity_msg.testFinished(test_name)
+
+    @staticmethod
+    @retrying.retry(
+        wait_fixed=(1 * 1000),
+        stop_max_delay=(120 * 1000),
+        retry_on_result=lambda x: not x)
+    def wait_for_dns(dcos_api, hostname):
+        """Return True if Mesos-DNS has at least one entry for hostname."""
+        hosts = dcos_api.get('/mesos_dns/v1/hosts/' + hostname).json()
+        return any(h['host'] != '' and h['ip'] != '' for h in hosts)
+
+    def setup_cluster_workload(self, dcos_api: DcosApiSession, healthcheck_app: dict, dns_app: dict):
+        """Start a cluster workload on entry and verify its health on exit."""
```

Comment:
The verification part of this docstring is no longer true. The method's name describes it pretty well, so we could just get rid of the docstring.